Error running vcf2xg.sh on human data

I've returned to give MoMI-G another shot, this time with human data but I get an error trying to generate the `xg` file.


```
$ ~/MoMI-G$ sudo bash MoMI-G/scripts/vcf2xg.sh proband_trim_bwaMEM_sort_dedupped.bam.generator.V2.overlap.hashcount.fastq.bam.FINAL.vcf momig-vcf-test-output MoMI-G/scripts/vg Homo_sapiens.GRCh37.dna.toplevel.fa


+ input=.//momig-vcf-test-output.pcf
+ reference=Homo_sapiens.GRCh37.dna.toplevel.fa
+ pref=.//momig-vcf-test-output.pcf
+ sort -t, -k 7n .//momig-vcf-test-output.pcf
+ sed -e 's/[ ]*//g'
+ awk -F '[,:]' '{print $1,"\t",$2,"\n",$4,"\t",$5}' /dev/fd/63
+ sort -k 1,1 -k 2,2n
+ uniq
++ sed 1d .//momig-vcf-test-output.pcf
+ '[' '!' -s Homo_sapiens.GRCh37.dna.toplevel.fa.fai ']'
++ dirname MoMI-G/scripts/vcf2gfa/pcf2gfa.sh
+ ruby MoMI-G/scripts/vcf2gfa/json_to_breakpoint_list.rb .//momig-vcf-test-output.pcf.bp.tsv Homo_sapiens.GRCh37.dna.toplevel.fa
+ cat .//momig-vcf-test-output.pcf.bp.tsv breakpoint_list_tmp.tsv
+ sort -k 1,1 -k 2,2n
+ uniq
++ dirname MoMI-G/scripts/vcf2gfa/pcf2gfa.sh
+ ruby MoMI-G/scripts/vcf2gfa/gfa_generator.rb .//momig-vcf-test-output.pcf.bp.merged.tsv .//momig-vcf-test-output.pcf.output.pcf Homo_sapiens.GRCh37.dna.toplevel.fa
Traceback (most recent call last):
	5: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:42:in `<main>'
	4: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:42:in `open'
	3: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:43:in `block in <main>'
	2: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:43:in `each_line'
	1: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:49:in `block (2 levels) in <main>'
MoMI-G/scripts/vcf2gfa/gfa_generator.rb:25:in `fasta': [ERROR] unexpected sequence :0- (RuntimeError)
	5: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:42:in `<main>'
	4: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:42:in `open'
	3: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:43:in `block in <main>'
	2: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:43:in `each_line'
	1: from MoMI-G/scripts/vcf2gfa/gfa_generator.rb:48:in `block (2 levels) in <main>'
MoMI-G/scripts/vcf2gfa/gfa_generator.rb:51:in `rescue in block (2 levels) in <main>': [ERROR] in 1 0 : [ERROR] unexpected sequence :0- (RuntimeError)
```

This is the output files:

```
(base) mtg@mtg-ThinkPad-P53:~/MoMI-G$ ls -al
total 31834880
drwxrwxr-x  4 mtg  mtg         4096 Nov 12 10:40 ./
drwx------ 78 mtg  mtg        24576 Nov 11 14:57 ../
-rw-r--r--  1 root root    14029435 Nov 12 10:40 breakpoint_list_tmp.tsv
-rw-rw-r--  1 mtg  mtg          798 Nov 11 15:04 config.yaml
-rwxr-x---  1 mtg  mtg  32570478447 Nov 11 21:48 Homo_sapiens.GRCh37.dna.toplevel.fa*
-rwxr-x---  1 mtg  mtg        11646 Nov 12 09:04 Homo_sapiens.GRCh37.dna.toplevel.fa.fai*
drwxrwxr-x  9 mtg  mtg         4096 Nov 11 15:06 MoMI-G/
-rw-r--r--  1 root root          11 Nov 12 10:40 momig-vcf-test-output.ggf
-rw-r--r--  1 root root         111 Nov 12 10:40 momig-vcf-test-output.json
-rw-r--r--  1 root root       13492 Nov 12 10:40 momig-vcf-test-output.pcf
-rw-r--r--  1 root root    14032093 Nov 12 10:40 momig-vcf-test-output.pcf.bp.merged.tsv
-rw-r--r--  1 root root        2658 Nov 12 10:40 momig-vcf-test-output.pcf.bp.tsv
-rw-r--r--  1 root root       13492 Nov 12 10:40 momig-vcf-test-output.pcf.output.pcf
-rw-r--r--  1 root root         558 Nov 12 10:40 momig-vcf-test-output.vg
-rw-r--r--  1 root root        3330 Nov 12 10:40 momig-vcf-test-output.xg
-rw-r--r--  1 mtg  mtg        62805 Nov 11 15:12 proband_trim_bwaMEM_sort_dedupped.bam.generator.V2.overlap.hashcount.fastq.bam.FINAL.vcf
-rw-r--r--  1 root root       62805 Nov 12 10:40 proband_trim_bwaMEM_sort_dedupped.bam.generator.V2.overlap.hashcount.fastq.bam.FINAL.vcf.vcf
```

Thank you so much. I re-checked and changed input file and vg file path, and re-ran it.
And now I got new errors. Are these vcf problems in my input ?
```
$ bash scripts/vcf2xg.sh test2.chr1.vcf test2_output ../vg/bin/vg hg19                                                
+ input=.//test2_output.pcf                                              
+ reference=hg19                                                         
+ pref=.//test2_output.pcf                                               
+ sort -t, -k 7n .//test2_output.pcf                                     
+ sed -e 's/[ ]*//g'                                                     
+ awk -F '[,:]' '{print $1,"\t",$2,"\n",$4,"\t",$5}' /dev/fd/63          
++ sed 1d .//test2_output.pcf                                            
+ sort -k 1,1 -k 2,2n                                                    
+ uniq                                                                   
+ '[' '!' -s hg19.fai ']'                                                
+ '[' '!' -s hg19 ']'                                                    
+ '[' '!' -s hg19.fa ']'                                                 
+ samtools faidx hg19.fa                                                 
+ reference=hg19.fa                                                      
++ dirname scripts/vcf2gfa/pcf2gfa.sh                                    
+ ruby scripts/vcf2gfa/json_to_breakpoint_list.rb .//test2_output.pcf.bp.tsv hg19.fa                                                              
+ cat .//test2_output.pcf.bp.tsv breakpoint_list_tmp.tsv                 
+ sort -k 1,1 -k 2,2n                                                    
+ uniq                                                                   
++ dirname scripts/vcf2gfa/pcf2gfa.sh                                    
+ ruby scripts/vcf2gfa/gfa_generator.rb .//test2_output.pcf.bp.merged.tsv .//test2_output.pcf.output.pcf hg19.fa                                  
Traceback (most recent call last):                                               
5: from scripts/vcf2gfa/gfa_generator.rb:42:in `<main>'                  
4: from scripts/vcf2gfa/gfa_generator.rb:42:in `open'                    
3: from scripts/vcf2gfa/gfa_generator.rb:43:in `block in <main>'         
2: from scripts/vcf2gfa/gfa_generator.rb:43:in `each_line'               
1: from scripts/vcf2gfa/gfa_generator.rb:49:in `block (2 levels) in <main>'                                                               
scripts/vcf2gfa/gfa_generator.rb:25:in `fasta': [ERROR] unexpected sequence :0- (RuntimeError)                                                            
5: from scripts/vcf2gfa/gfa_generator.rb:42:in `<main>'                  
4: from scripts/vcf2gfa/gfa_generator.rb:42:in `open'                    
3: from scripts/vcf2gfa/gfa_generator.rb:43:in `block in <main>'         
2: from scripts/vcf2gfa/gfa_generator.rb:43:in `each_line'               
1: from scripts/vcf2gfa/gfa_generator.rb:48:in `block (2 levels) in <main>'                                                               
scripts/vcf2gfa/gfa_generator.rb:51:in `rescue in block (2 levels) in <main>': [ERROR] in chr1 0 : [ERROR] unexpected sequence :0- (RuntimeError)
```

Thank you for your instruction.
Then I git pulled and re-tried. I checked the VCF file and hg19.fa and confirmed that chr1 (coordinate)  is the format in them. So this is not the case that one is "chr1" and the other is "1".
Anyway, I hope the notified errors I experienced this time will help to solve the problems.
```
$ bash scripts/vcf2xg.sh test.chr1.vcf test3_output ../vg/bin/vg hg19
+ input=.//test3_output.pcf
+ reference=hg19
+ pref=.//test3_output.pcf
+ sort -t, -k 7n .//test3_output.pcf
+ sed -e 's/[ ]*//g'
+ awk -F '[,:]' '{print $1,"\t",$2,"\n",$4,"\t",$5}' /dev/fd/63
+ sort -k 1,1 -k 2,2n
++ sed 1d .//test3_output.pcf
+ uniq
+ '[' '!' -s hg19.fai ']'
+ '[' '!' -s hg19 ']'
+ '[' '!' -s hg19.fa ']'
+ samtools faidx hg19.fa
+ reference=hg19.fa
++ dirname scripts/vcf2gfa/pcf2gfa.sh
+ ruby scripts/vcf2gfa/json_to_breakpoint_list.rb .//test3_output.pcf.bp.tsv hg19.fa
+ cat .//test3_output.pcf.bp.tsv breakpoint_list_tmp.tsv
+ sort -k 1,1 -k 2,2n
+ uniq
++ dirname scripts/vcf2gfa/pcf2gfa.sh
+ ruby scripts/vcf2gfa/gfa_generator.rb .//test3_output.pcf.bp.merged.tsv .//test3_output.pcf.output.pcf hg19.fa
Traceback (most recent call last):
	5: from scripts/vcf2gfa/gfa_generator.rb:42:in `<main>'
	4: from scripts/vcf2gfa/gfa_generator.rb:42:in `open'
	3: from scripts/vcf2gfa/gfa_generator.rb:43:in `block in <main>'
	2: from scripts/vcf2gfa/gfa_generator.rb:43:in `each_line'
	1: from scripts/vcf2gfa/gfa_generator.rb:51:in `block (2 levels) in <main>'
scripts/vcf2gfa/gfa_generator.rb:25:in `fasta': unexpected sequence :0- (RuntimeError)
	5: from scripts/vcf2gfa/gfa_generator.rb:42:in `<main>'
	4: from scripts/vcf2gfa/gfa_generator.rb:42:in `open'
	3: from scripts/vcf2gfa/gfa_generator.rb:43:in `block in <main>'
	2: from scripts/vcf2gfa/gfa_generator.rb:43:in `each_line'
	1: from scripts/vcf2gfa/gfa_generator.rb:50:in `block (2 levels) in <main>'
scripts/vcf2gfa/gfa_generator.rb:53:in `rescue in block (2 levels) in <main>': [ERROR] in 'chr1 0' : unexpected sequence :0- (RuntimeError)
```

--- a/scripts/vcf2gfa/gfa_generator.rb
+++ b/scripts/vcf2gfa/gfa_generator.rb
@@ -22,7 +22,7 @@ end
 
 def fasta(current_read, start, stop)
   seq = "#{current_read}:#{start}-#{stop}"
-  raise "unexpected sequence #{seq}" if !current_read || !start || !stop
+  raise "unexpected genomic range #{seq}" if !current_read || !start || !stop
   fasta = `samtools faidx #{REF} #{seq}`
   fasta
 end
@@ -43,10 +43,10 @@ File.open(ARGV[0]) do |f|
   f.each_line do |line|
     line = line.chomp.split(" ")
     next if line[0].end_with?("id")
-    line_orig = line
+    line_orig = line.clone
     line[1] = line[1].to_i
     raise "ERROR: Reference id '#{line[0]}' is not found in #{REF}" if !ref_len[line[0]]
-    if line[1]==0 # When reads start from 0
+    if line[1]==0 && current_read != "" # When reads start from 0
       begin
         fasta = fasta(current_read, prev_pos, ref_len[current_read])
       rescue => exception

